### PR TITLE
HF Referral Data Model and Repositories changes

### PR DESCRIPTION
**Pull request by Ramkrishna-egov**

### CODE REVIEW

File: packages/referral_reconciliation/lib/pages/search_referral_reconciliations.dart [hunk 1]
```diff
@@ -56,26 +66,35 @@ class _SearchReferralReconciliationsPageState
...
-      genderOptions: widget.genders,
-      referralReasons: widget.referralReasons,
-      cycles: widget.cycles,
-      checklistTypes: widget.checklistTypes,
+    searchReferralsBloc = SearchReferralsBloc(
+      const SearchReferralsState(),
+      referralReconDataRepository:
+          context.repository<HFReferralModel, HFReferralSearchModel>(context),
     );
+    // ReferralReconSingleton().setInitialData(
```
Comment (naveenr-egov) [inline packages/referral_reconciliation/lib/pages/search_referral_reconciliations.dart:78]: remove commented code

File: apps/health_campaign_field_worker_app/lib/blocs/referral_reconciliation/hcm_referral_reconciliation_bloc.dart [hunk 2]
```diff
@@ -61,7 +45,7 @@ class HcmReferralReconBloc extends ReferralReconListener {
 
   // Fetch project facilities for project ID
   @override
-  Future<List<ReferralProjectFacilityModel>>
+  Future<List<ProjectFacilityModel>>
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/referral_reconciliation/hcm_referral_reconciliation_bloc.dart:48]: throw unimplemented error if it's not being used

File: packages/digit_scanner/lib/pages/qr_scanner.dart [hunk 3]
```diff
@@ -75,426 +75,402 @@ class _DigitScannerPageState extends LocalizedState<DigitScannerPage> {
...
-                                  border: Border.all(
-                                    color: DigitTheme
-                                        .instance.colorScheme.outline,
-                                    width: 1,
+
+                        Positioned(
+                          bottom: (kPadding * 7.5),
+                          height: widget.isGS1code
+                              ? state.barCodes.length < 10
+                                  ? (state.barCodes.length * 60) + 80
```
Comment (naveenr-egov) [inline packages/digit_scanner/lib/pages/qr_scanner.dart:253]: change to mediaquery

Comment (Ramkrishna-egov) [reply packages/digit_scanner/lib/pages/qr_scanner.dart:253]: Its based on the barcode length , not the mediaquery size

File: apps/health_campaign_field_worker_app/lib/pages/home.dart [hunk 4]
```diff
@@ -488,6 +479,10 @@ class _HomePageState extends LocalizedState<HomePage> {
                               ?.map((e) => e.code)
                               .toList() ??
                           [],
+                      boundaryCode: context.boundary.code ?? '',
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/pages/home.dart:482]: If data being set in singleton method remove in navigation args

File: packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart [hunk 5]
```diff
@@ -340,7 +391,7 @@ class _RecordReferralDetailsPageState
                                                     );
                                                 final parent = context.router
```
Comment (naveenr-egov) [inline packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart:392]: parent route is been assigned but not used remove if not required

File: packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart [hunk 6]
```diff
@@ -462,7 +513,7 @@ class _RecordReferralDetailsPageState
                                                     final parent =
                                                         context.router.parent()
                                                             as StackRouter;
-                                                    parent.push(
+                                                    context.router.push(
```
Comment (naveenr-egov) [inline packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart:516]: remove parent if not used

**Review by naveenr-egov — CHANGES_REQUESTED** [review]

**Review by naveenr-egov — APPROVED** [review]

**Pull request merged**